### PR TITLE
chore: sphinx documentation configs, build script, CI job

### PR DESCRIPTION
**Pull request by lost-theory**
### What I did

Related issue: #21 

### How I did it

I used some of the sphinx setup I did in eip712, but added quite a bit of work on top of that, including the [sphinx-autoapi](https://pypi.org/project/sphinx-autoapi/) plugin so we don't have to generate or maintain `.rst` files for full API documentation of modules, classes, and functions.

### How to verify it

A new CI job for docs will on PRs, branches, and releases.

### Checklist

- [x] Passes all linting checks (pre-commit and CI jobs)
- [x] New test cases have been added and are passing
- [x] Documentation has been updated
- [x] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

File: src/ape/types/__init__.py [hunk 1]
```diff
@@ -3,6 +3,14 @@
...
 from .manifest import PackageManifest, PackageMeta
 
+
+class Foo:
+    """The Foo class."""
+
+    def __init__(self):
+        pass
+
+
```
Comment (fubuloubu) [inline src/ape/types/__init__.py:13]: Don't need this

File: src/ape/managers/networks.py [hunk 2]
```diff
@@ -12,29 +12,23 @@
...
 
-    When there are multiple providers in use, you must specify which to work with:
+    When there are multiple providers in use, you must specify which to work with::
 
-    ```
-    eth_contract = Contract.at(network=eth_mainnet)
-    bsc_contract = account.deploy(Contract, network=bsc_mainnet)
-    ```
+        eth_contract = Contract.at(network=eth_mainnet)
+        bsc_contract = account.deploy(Contract, network=bsc_mainnet)
```
Comment (fubuloubu) [inline src/ape/managers/networks.py:31]: I don't know if this code functions this way since it's a context manager

File: docs/quickstart.rst [hunk 3]
```diff
@@ -0,0 +1,69 @@
+Quickstart guide
```
Comment (fubuloubu) [inline docs/quickstart.rst:1]: I wonder if there is a way to pull this content from the `README.md`?

File: src/ape/__init__.py [hunk 4]
```diff
@@ -34,19 +37,36 @@
...
+
 project = Project(config.PROJECT_FOLDER)
+"""The currently active project. See :class:`ape.managers.project.ProjectManager`."""
 
 Contract = _partial(_Contract, networks=networks, converters=_converters)
+"""User-facing class for instantiating contracts. See :class:`ape.api.contracts._Contract`."""
 
 convert = _converters.convert
+"""Conversion utilities for the active project. See
+:class:`ape.managers.converters.ConversionManager`."""
```
Comment (fubuloubu) [inline src/ape/__init__.py:69]:
```suggestion
"""Conversion utility function. See :class:`ape.managers.converters.ConversionManager`."""
```

File: docs/quickstart.rst (hunk 3 above)
Comment (lost-theory) [reply docs/quickstart.rst:1]: We could add a step to `build_docs.py` to do that. It would make things easier if we converted it from markdown to ReST.

**Review by fubuloubu — APPROVED** [review]

**Pull request merged**